ocs_ci/ocs/platform_nodes.py

```diff
@@ -432,3 +467,424 @@ def restart_nodes_teardown(self):
+                user=constants.EC2_USER
+            )
+            rhel_pod_obj.copy_to_server(
+                host, pem_dst_path,
+                os.path.join(dst, constants.INTERNAL_MIRROR_PEM_FILE),
+                os.path.join('/tmp', constants.INTERNAL_MIRROR_PEM_FILE),
+                user=constants.EC2_USER,
+            )
+            cmd = (
+                f'sudo mv '
+                f'{os.path.join("/tmp/", constants.INTERNAL_MIRROR_PEM_FILE)} '
+                f'{dst}'
+            )
+            rhel_pod_obj.exec_cmd_on_node(
+                host, pem_dst_path,
+                cmd, user=constants.EC2_USER
+            )
+        # copy kubeconfig to pod
+        kubeconfig = os.path.join(
+            self.cluster_path, config.RUN.get('kubeconfig_location')
+        )
+        pod.upload(rhel_pod_obj.name, kubeconfig, "/")
+        pull_secret_path = os.path.join(
+            constants.TOP_DIR,
+            "data",
+            "pull-secret"
+        )
+        pod.upload(rhel_pod_obj.name, pull_secret_path, "/tmp/")
+        host_file = self.build_ansible_inventory(hosts)
+        pod.upload(rhel_pod_obj.name, host_file, "/")
+        # install pod packages
+        rhel_pod_obj.install_packages(constants.RHEL_POD_PACKAGES)
+        # run ansible
+        cmd = (
+            f"ansible-playbook -i /hosts --private-key={pem_dst_path} "
+            f"{constants.SCALEUP_ANSIBLE_PLAYBOOK}"
+        )
+
+        rhel_pod_obj.exec_cmd_on_pod(
+            cmd, out_yaml_format=False, timeout=timeout
+        )
+        self.verify_nodes_added(hosts)
+
+    def verify_nodes_added(self, hosts):
+        """
+        Verify RHEL workers are added
+
+        Args:
+             hosts (list): list of aws private hostnames
+
+        Raises:
+            FailedToAddNodeException: if node addition failed
+
+        """
+        timeout = 600
+        ocp_obj = ocp.OCP(kind='node')
+        node_info = ocp_obj.get()
+        for i in range(len(hosts)):
+            for entry in node_info['items']:
+                for each in entry['status']['addresses']:
+                    if each['type'] == 'Hostname':
+                        if each['address'] in hosts:
+                            logging.info(
+                                f"Checking status for {each['address']}"
+                            )
+                            sample = TimeoutSampler(
+                                timeout, 3,
+                                self.get_ready_status, entry
+                            )
+                            try:
+                                assert sample.wait_for_func_status(result=True)
+                            except AssertionError:
+                                raise exceptions.FailedToAddNodeException(
+                                    "Failed to add RHEL node"
+                                )
+
+    def get_ready_status(self, node_ent):
```

Comment:
can we reuse https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/aws.py#L742

Reply:
Actually this function will live in this file and we will remove it from deployment/aws.py. All the node related stuff will come to this file.